csrc/index_compute.cpp

```diff
@@ -3291,81 +3291,598 @@ Val* Index::eye(
+// (0, 2), (0, 4), ..., (2, 0), (2, 2), (2, 4), ..., (6, 6). We will transfer
+// data as (4, 4) dense tile of elements. In this scenario, the `elementStrides`
+// in  `cuTensorMapEncodeTiled` will have a value [2, 2]. For this case, the
+// originating bulk IterDomains must be the outer of the split that defines it,
+// where the inner of this split must be non-bulk. The parent of the originating
+// bulk IterDomains must also be an output of a split and it must be the inner
+// of the split that defines it, and similarly, the outer of this split must be
+// non-bulk. For this case, the extents of the originating bulk IterDomains are
+// the tile sizes, and the extents of parents of the originating bulk
+// IterDomains are the box sizes (the `boxDim` parameter in
+// `cuTensorMapEncodeTiled`).
+//
+// With this understanding, we can now define more terminologies:
+//
+// Definition 3: "boxId" is a function that takes an originating bulk IterDomain
+// as its input, and returns an IterDomain. Given an originating bulk IterDomain
+// I1, the return value, boxId(I1) is:
+//   - I1 if I1 is the inner of the split that defines it
+//   - The parent of I1 if I1 is the outer of the split that defines it
+//
+// In the above diagram, boxId(I4) is I4, and boxId(I7) is I6.
+//
+// Definition 4: "globalId" is a function that takes an originating bulk
+// IterDomain as its input, and returns another IterDomain. Given an originating
+// bulk IterDomain I1, the return value, globalId(I1) is:
+//   - The parent of I1 if I1 is the inner of the split that defines it
+//   - The grandparent of I1 if I1 is the outer of the split that defines it
+//
+// In the above diagram, globalId(I4) is I1, and globalId(I7) is I2.
+//
+// Definition 5: "innerId" is a function that takes an originating bulk who is
+// the outer of the split that defines it, and returns its sibling IterDomain.
+//
+// In the above diagram, innerId(I7) is I8.
+//
+// Definition 6: An IterDomain I1 is "global" if there exists an originating
+// bulk IterDomain I2 such that I1 == globalId(I2).
+//
+// In the above diagram, I1 and I2 are global IterDomains.
+//
+// Note that the number of global IterDomains should always be equal to the
+// number of originating bulk IterDomains, which should always be equal to the
+// dimensionality of the tensor in the eyes of TMA.
+//
+// Also note that global IterDomains can not have dependencies between each
+// other, that is, no one global IterDomains can be an ancestor or descendant of
+// another global IterDomains. That is, schedules like this is not allowed:
+//   I1 -> split -> Bulk (inner)
+//              \-> I2 (outer) -> split -> Bulk (inner)
+//                                     \-> I3 (outer)
+// Also note that the set of all global IterDomains must be "equivalent" to the
+// allocation domain of the tensor (ignoring unrelated IterTypes like broadcast
+// and reduction), as defined in `ir_utils::validateDomainEquivalence`.
```

Comment:
A minor suggestion, given the number of new terms introduced in this comment, I would find it really useful to see a set of examples with asii-art diagrams followed by explanation which IterDomains are originating bulk, box, global - and what will be passed to `cuTensorMapEncodeTiled`.

We could have examples for:
- 1D tensor - tiled to 2d and scheduled as 2d dense TMA transaction (box == tile)
- 1D tensor - tiled to 2d and scheduled as 2d sparse TMA transaction (box != tile)
- 2D tensor - tiled, but with merge expres included, scheduled as 2d dense TMA transaction

@zasdfgbnm , what do you think about this idea? Something similar to what I included in my [comment](https://github.com/NVIDIA/Fuser/pull/1962#discussion_r1533581104)

Reply:
Good suggestion! I will add examples.

Reply:
Added in https://github.com/NVIDIA/Fuser/pull/1962/commits/62dd8b9fb72872b21336b4db128892c7fb6dae1e